sycl/include/syclcompat/memory.hpp

```diff
@@ -888,7 +888,8 @@ template <class T, memory_region Memory, size_t Dimension> class device_memory {
     assert(init_list.size() <= in_range.size());
     _host_ptr = (value_t *)std::malloc(_size);
     std::memset(_host_ptr, 0, _size);
-    std::memcpy(_host_ptr, init_list.begin(), init_list.size() * sizeof(T));
+    sycl::detail::memcpy(_host_ptr, init_list.begin(),
```

Comment:
Looking at `std::malloc`, it seems like this is also a host-only API and therefore we can leave `std::memcpy` as-is